### PR TITLE
plumb broker annotations through, add docs

### PR DESCRIPTION
**Pull request by vaikas**
Previously the values for scaledobjects were hardcoded, now take the values from the broker annotations and handle lifecycle if annotations get modified or deleted.
Add documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,10 +15,11 @@
 To enable KEDA Autoscaling of Knative Event Sources (and other components in the future) there is a separate controller implemented, ie. no hard dependency in Knative.
 This contoller si watching for `CustomResourcesDefinitions` resources in the cluster, if there is installed a new CRD which is supported by this controller a new dynamic controller watching these resources is created. 
 
-Currently there is support for **Kafka Source** and **AWS SQS Source**.
+Currently there is support for **Kafka Source** and **AWS SQS Source**. We also
+have experimental support for **RabbitMQ Broker**.
 
 ## Annotations
-User can enable and configure autoscaling on a particular Source by a set of annotations. 
+User can enable and configure autoscaling on a particular Source or Broker by a set of annotations.
```
Comment (mattmoor) [inline README.md:22]: Remove trailing whitespace:
```suggestion
User can enable and configure autoscaling on a particular Source or Broker by a set of annotations.
```

File: README.md [hunk 2]
```diff
@@ -128,4 +126,30 @@ spec:
 $ kubectl get scaledobjects
 NAME                                      SCALETARGETKIND      SCALETARGETNAME                                                 TRIGGERS   AUTHENTICATION   READY   ACTIVE   AGE
 so-f87369e5-c320-4f44-b23a-8c535a523e3a   apps/v1.Deployment   kafkasource-kafka-source-f87369e5-c320-4f44-b23a-8c535a523e3a   kafka                       True    False     6m5s
-```
\ No newline at end of file
+```
+
+## Example of RabbitMQ Broker autoscaled by KEDA
+
+1. Install Knative Serving and Eventing
```
Comment (mattmoor) [inline README.md:133]: Remove trailing whitespace:
```suggestion
1. Install Knative Serving and Eventing
```

File: README.md (hunk 1 above)
Comment (mattmoor) [inline README.md:22]: Remove trailing whitespace:
```suggestion
User can enable and configure autoscaling on a particular Source or Broker by a set of annotations.
```

File: README.md [hunk 3]
```diff
@@ -128,4 +122,30 @@ spec:
...
```
Comment (mattmoor) [inline README.md:129]: Remove trailing whitespace:
```suggestion
1. Install Knative Serving and Eventing
```

**Review by n3wscott — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**